src/data/astarte_message_hub_node.rs

```diff
@@ -0,0 +1,734 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2023 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Contains the implementation for the Astarte message hub node.
+
+use astarte_message_hub::proto_message_hub;
+use async_trait::async_trait;
+use log::warn;
+use serde::Deserialize;
+use std::time::Duration;
+use tonic::Response as TonicResponse;
+use tonic::Streaming as TonicStreaming;
+
+use crate::data::{Publisher, Subscriber};
+use crate::error::DeviceManagerError;
+use crate::DeviceManagerOptions;
+
+/// Device runtime node identifier.
+const DEVICE_RUNTIME_NODE_UUID: &str = "d72a6187-7cf1-44cc-87e8-e991936166db";
+
+/// Struct containing the configuration options for the Astarte message hub.
+#[derive(Debug, Deserialize, Clone)]
+pub struct AstarteMessageHubOptions {
+    /// The Endpoint of the Astarte Message Hub
+    endpoint: String,
+}
+
+/// Struct containing the configuration options for the Astarte message hub node.
+pub struct AstarteMessageHubNodeOptions {
+    /// The Endpoint of the Astarte Message Hub
+    endpoint: String,
+    /// Directory containing the Astarte interfaces.
+    interfaces_directory: String,
+}
```

Comment:
These two structs seem redundant.
The only reason for the existence of `AstarteMessageHubNodeOptions` is to provide the `new` function with the `interfaces_directory` right?
Can't we reach the same goal by passing the `interfaces_directory` as a separate argument?